src/check/ast.rs

```diff
@@ -0,0 +1,30 @@
+use std::ops::Deref;
+
+use crate::check::name::Name;
+use crate::common::position::Position;
+use crate::parse::ast::{AST, Node};
+
+#[derive(PartialEq, Eq, Hash, Clone, Debug)]
+pub struct ASTTy {
+    pub pos: Position,
+    pub node: Node,
```

Comment:
Need new `Node` variant which has `ASTTy` children instead of `AST` children.
Going to be a lot of work, probably for a next PR.
This PR only changes the interface for now.